gematria/datasets/annotating_importer.cc

```diff
@@ -290,7 +280,24 @@ AnnotatingImporter::GetLBRBlocksWithLatency() {
   if (!elf_object.ok()) {
     return elf_object.status();
   }
-  const auto main_header = GetMainProgramHeader(*elf_object);
+
+  // TODO(vbshah): Consider making it possible to use other ELFTs rather than
+  // only ELF64LE since only the implementation of GetMainProgramHeader differs
+  // between different ELFTs.
+  if (!(*elf_object)->is64Bit() || !(*elf_object)->isLittleEndian()) {
```

Comment:
I figured that would not be necessary since `GetBlocksFromELF` is to be removed soon, since it duplicates the functionality in `extract_bbs_from_obj_lib` (except *maybe* some instruction addressing specifics - apart from `getBasicBlockHexValues` not returning anything with addresses, `GetBlocksFromELF` was required to use instruction addresses matching exactly with the corresponding sample addresses, which is probably useful to have).

Reply:
If it's going to get removed soon, this seems reasonable enough to me.